### ISSUE TITLE
Edit show is broken, at least for Anime

### ISSUE THREAD

**Issue by hergonoway** [pymedusa/Medusa#4633]
Version : 0.2.6 (through docker)
Basic stuff tried : Ive tried to reboot, rebuild the container from the last image, same result.

TL;DR : After adding a new anime show, if I edit the show, parameters are set to something else, and if corrected and saved, File location is lost.

Step 1 : I've added a new anime show, with "HD1080p" as quality and anime and scene numbering options, and unchecked folder for seasons  (like 20 sec before the screen shot) I didn't choose to whitelist a release group since it's currently bugged (in another issue) : 
<img width="587" alt="medusa 2018-07-08 00-35-28" src="https://user-images.githubusercontent.com/8405448/42415193-eaf2973c-8246-11e8-9c64-c02090e52e94.png">

 Step2 : I click on "Edit" on the show page : 
> On tab "MAIN" :  there "HD1080p" is replaced by "Custom" without any quality option checked, "default status episode" is empty. Also, the name of the Show is missing. 
> On tab "Format" : almost everything is wrong 
<img width="410" alt="edit show 2018-07-08 00-36-55" src="https://user-images.githubusercontent.com/8405448/42415202-127860de-8247-11e8-9d3a-10482a3524f9.png">

<img width="471" alt="edit show 2018-07-08 00-37-34" src="https://user-images.githubusercontent.com/8405448/42415205-2f18c1b6-8247-11e8-92a0-e975c08f3114.png">

Step3 : once I've corrected everything and click "Save", I go back to the show page and now Location is missing ...
<img width="389" alt="medusa - isekai maou to shoukan shoujo no dorei majutsu 2018-07-08 00-38-57" src="https://user-images.githubusercontent.com/8405448/42415208-57758068-8247-11e8-9570-a8ffd732c667.png">

I've tried it with 3 different anime, same result.

Bonus : if you can  whitelisted release group while adding the show (if you can because it's currently with the old UX which is bugged), for exemple "HorribleSubs" you could end up like me with "H","o","r","i","b","l","e","S","u","b","s" :)

**Comment by p0psicles** [comment]
Any errors in browsers console?
Also maybe a slow connection? Like you sure you gave it the time to load?

**Comment by hergonoway** [comment]
Hello, thanks for your quick answer. I've tried to reproduced the problem, but something changed from yesterday. 

Yesterday I tried to add the following show : 
- Shichisei no Subaru (Main title) ; Seven Senses of the Re`Union (Official Title)
- Isekai Maou to Shoukan Shoujo no Dorei Majutsu (Maint title) ; How Not to Summon a Demon Lord (official title)

At the time, only TMDB indexer returned something for both show and only with the Main Title, Official title didn't returned anything on all indexer. And problems where encountered from there on these added shows with TMDB.

Today these 2 shows doesn't give result on Maint title anymore, but TMDB return something on the Official title, and now TVDB and TVMaze return something also.

From now the problem doesn't show up anymore, in the meantime I did'nt restart the server or anything else, tested and retested the issue on Safari, Firefox and Chrome like yesterday.

It a very strange behavior, do you think that maybe these show where too fresh (in indexer repository) and since there's obvious change made on TMDB between yesterday and today (It doesnt respond to the same title) it could had an impact on Medusa while it tries to build up the show from TMDB infos ?

**Comment by p0psicles** [comment]
I really dont know. Hard to say if we cant reproduce it.

On a sidenote i would only use tvdb. The searching makes heavy use of xem. And that only works with tvdb.

**Comment by sharkykh** [comment]
Last time we had an issue like this it was a bad character encoding on our end (#4248, #4260).
@hergonoway 
You didn't get any errors in the logs?

**Comment by gilfrade** [comment]
I have the same version and this is also happening to me.

1º Added New Show without setting "Anime"  
2º Edit works!  
3º Edited Show to "Anime" and Saved  
4º Edit does not works  

editShow endpoint is pending without response.

**Comment by sharkykh** [comment]
@gilfrade 
Which show is that?
I just tried `Seven Senses of the Re'Union` from TMDB with your steps and it works for me?
Also please check the browser's developer console (<kbd>F12</kbd> key or <kbd>CTRL+SHIFT+I</kbd>) for errors.

**Comment by alron** [comment]
Seeing the exact same problem with 'Yume Oukoku to Nemureru 100-nin no Ouji-sama' (100 Sleeping Princes & The Kingdom of Dreams). Added as anime, the edit button doesn't present a usable page.

Have deleted and added multiple times. Full updated multiple times as well.

![screenshot 2018-07-24 11 56 57](https://user-images.githubusercontent.com/566136/43160012-d2d90722-8f38-11e8-880d-c61afc33e136.png)

**Comment by sharkykh** [comment]
I'm still unable to reproduce this.
@alron 
Please check the browser's developer console (F12 key or CTRL+SHIFT+I) for errors.
If there aren't any, check the Network tab, refresh the page and then search for `/api/v2/series`, and post the response it got here.

**Comment by alron** [comment]
@sharkykh 
The only error showing on edit page is a 500 series error. Same error on network console (of course).
![100 sleeping princes - edit show console error](https://user-images.githubusercontent.com/566136/43215908-48d3df60-8ff2-11e8-9dfe-fd6f1a0c4252.png)

Seeing a 404 error on the show page itself. But this appears to be for the banner.
![100 sleeping princes - display show console error](https://user-images.githubusercontent.com/566136/43215805-069f68b2-8ff2-11e8-81fb-18cf090574c2.png)

**Comment by p0psicles** [comment]
Ok then we really need some debug logs at time of the 500 error

**Comment by alron** [comment]
@p0psicles @sharkykh 
Looks like an exception in the tornado side of things.

Errors didn't start showing up in debug log until I enabled logging for tornado.

Let me know if there is anything else I can provide.

[error.txt](https://github.com/pymedusa/Medusa/files/2229255/error.txt)

**Comment by sharkykh** [comment]
That's good to know.

I have a workaround for it, but it won't fix the underlying issue.

**Comment by sharkykh** [comment]
@alron If you don't mind, I'd like your help in finding the issue. It requires some code editing.

Open this file:
https://github.com/pymedusa/Medusa/blob/e712b058351c89c3d420c5e0901023d56afa1aed/medusa/tv/episode.py?utf8=%E2%9C%93#L1056

And replace that line with (note that the indentation is important):
```py
        try:
            data['airDate'] = self.air_date
        except Exception:
            ep_data = [
                'episode: ' + repr(data['identifier']),
                'show: ' + repr(data['id']),
                'airdate: ' + repr(self.airdate),
                'airs: ' + repr(self.series.airs),
                'network: ' + repr(self.series.network)
            ]
            log.exception('Failed to convert air date for:\n{0}', '\n'.join(ep_data))
```

Save the file, restart Medusa, and navigate to the edit show page again.
The page will now load correctly, and you should get errors in the server logs.
Please post them here.

If you can do this, it will help a lot in figuring out the issue.

**Comment by alron** [comment]
@sharkykh Roger that. I'll give it a whirl. Will post results shortly.

**Comment by alron** [comment]
@sharkykh Looks like that handled the exception a bit better and made the edit show page work.
![screenshot 2018-07-25 20 29 13](https://user-images.githubusercontent.com/566136/43240034-8692a9b0-9049-11e8-929f-50096e5ab402.png)

Log:
[error.txt](https://github.com/pymedusa/Medusa/files/2230351/error.txt)

**Comment by sharkykh** [comment]
@alron 
Thank you.
I'm still unable to reproduce this error, which makes it very hard for me to find the actual issue.
Maybe this is related - I'm running Medusa on Windows 7, what OS are you running?

**Comment by alron** [comment]
@sharkykh 
Docker 18.06.0~ce~3-0~ubuntu using the pymedusa/medusa image.
Base system docker lives on is Ubuntu Linux 18.04, kernel 4.15.0-24-generic. 
Using nginx as HTTPS proxy for the UI.

**Comment by sharkykh** [comment]
Huh, Docker too. Just like OP. Interesting.

I'll set up a Docker environment to test this.

In the meantime, if could you do one more code change please...
https://github.com/pymedusa/Medusa/blob/e7c4ff3bc9a48b47c932b0e40a539d62e3a62f64/medusa/tv/episode.py#L404-L410

Change these lines to:
```py
        airdate = date.toordinal(self.airdate)
        airtime = network_timezones.parse_date_time(airdate, self.series.airs, self.series.network)
        apptime = sbdatetime.convert_to_setting(airtime)
        return apptime.isoformat(b'T')
```

**Comment by sharkykh** [comment]
I still can't reproduce it, even inside a Docker container.

Could you change
```py
        apptime = sbdatetime.convert_to_setting(airtime)
        return apptime.isoformat(b'T')
```
to:
```py
        apptime = sbdatetime.convert_to_setting(airtime)
        log.warning('Airtime is: {0!r}', apptime)
        return apptime.isoformat(b'T')
```

Could you also check your timezone inside the container?
Maybe it's related...

**Comment by alron** [comment]
Timezone is set and reporting properly via cli utilities. 

Modified, restarted, triggered.

[error.txt](https://github.com/pymedusa/Medusa/files/2230592/error.txt)

**Comment by sharkykh** [comment]
I still think it's somehow related to timezone setting, because I don't get the error.
Can you please run the following command inside the container?
```py
python -c "import time; print time.timezone, time.altzone, time.tzname"
```

**Comment by alron** [comment]
```
root@a8874e476ce7:/$ python --version
Python 2.7.14
root@a8874e476ce7:/$ python -c "import time; print time.timezone, time.altzone, time.tzname"
28800 25200 ('PST', 'PDT')
```

**Comment by sharkykh** [comment]
Alright, now I'm getting somewhere:
I assume you set the timezone using (for example):
```shell
export TZ=America/Los_Angeles
```

I did that too. Then I did the following:
```
cd /app/medusa/ext
python -c "import datetime; from dateutil.tz import tzlocal; print datetime.datetime(1, 1, 1, 0, 0, tzinfo=tzlocal()).isoformat(b'T')"
```
Then I got the same error. Please try that on your container to make sure.

**Comment by alron** [comment]
```
root@a8874e476ce7:/$ cd /app/medusa/ext
root@a8874e476ce7:/app/medusa/ext$ python -c "import datetime; from dateutil.tz import tzlocal; print datetime.datetime(1, 1, 1, 0, 0, tzinfo=tzlocal()).isoformat(b'T')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "dateutil/tz/tz.py", line 210, in utcoffset
    if self._isdst(dt):
  File "dateutil/tz/tz.py", line 282, in _isdst
    if self.is_ambiguous(dt):
  File "dateutil/tz/tz.py", line 244, in is_ambiguous
    (naive_dst != self._naive_is_dst(dt - self._dst_saved)))
OverflowError: date value out of range
```

**Comment by alron** [comment]
![screenshot 2018-07-25 23 58 19](https://user-images.githubusercontent.com/566136/43246486-a399aff0-9066-11e8-9d5d-8457e8bf96c8.png)

Looks correct.

**Comment by sharkykh** [comment]
Try on the container:
```
cat /etc/localtime
```

**Comment by alron** [comment]
You very much so don't want me to cat /etc/localtime. It is binary TZ data.

**Comment by sharkykh** [comment]
Then `less`. I just want to make sure everything is set correctly.
What I'm trying to figure out is where a timezone setting is missing.

By the way for me, on the newly created container `/etc/localtime` is UTC, so it's small.
```
root@328005d0485e:/$ cat /etc/localtime
TZif2UTCTZif2UTC
UTC0
```

I don't use Linux regularly so sorry if I'm being meticulous with my checks.

**Comment by sharkykh** [comment]
Ok, nevermind. I think I know why it doesn't work.

**Comment by alron** [comment]
I'm pretty positive I have my dockers all set up properly. :)

My localtime is a docker filesystem mount of my host system /usr/share/zoneinfo/America/Los_Angeles (via the /usr/share/zoneinfo/US/Pacific symlink).

Have the first and last lines of the file. Pretty sure the last line is what you care about anyways.
```
root@a8874e476ce7:/$ strings /etc/localtime
TZif2
...
PST8PDT,M3.2.0,M11.1.0
```

**Comment by sharkykh** [comment]
@alron Thank you for your help. I was able to figure it out.

Some or most of the active developers here live on the eastern hemisphere with positive timezones, that's why it was so hard to reproduce.
Basically it was trying to subtract your timezone's UTC offset - (-7) hours, from the smallest possible datetime (year 1, month 1, day 1, hour 0, minute 0, etc...), that's why it was causing an "overflow" issue.
For example I live in Israel where the current timezone is UTC+3, so I was **adding** to it...

Thank you again :)

**Comment by alron** [comment]
@sharkykh That makes perfect sense. Can't go before epoch.
